src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -249,21 +249,23 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         this.modelingExercise = undefined;
         this.exampleSolution = undefined;
         this.exampleSolutionUML = undefined;
-        this.isProgrammingExerciseExampleSolutionPublished = false;
 
-        if (newExercise.type === ExerciseType.MODELING) {
-            this.modelingExercise = newExercise as ModelingExercise;
-            if (this.modelingExercise.exampleSolutionModel) {
-                this.exampleSolutionUML = JSON.parse(this.modelingExercise.exampleSolutionModel);
-            }
-        } else if (newExercise.type === ExerciseType.TEXT || newExercise.type === ExerciseType.FILE_UPLOAD) {
-            const exercise = newExercise as TextExercise & FileUploadExercise;
-            if (exercise.exampleSolution) {
-                this.exampleSolution = this.artemisMarkdown.safeHtmlForMarkdown(exercise.exampleSolution);
-            }
-        } else if (newExercise.type === ExerciseType.PROGRAMMING) {
-            const exercise = newExercise as ProgrammingExercise;
-            this.isProgrammingExerciseExampleSolutionPublished = exercise.exampleSolutionPublished || false;
+        switch (newExercise.type) {
```

Comment:
Please no switch cases

Reply:
Is there any reason for your dislike of switch cases?

Reply:
> It’s hard to make a small switch statement.6 Even a switch statement with only two cases is
> larger than I’d like a single block or function to be. It’s also hard to make a switch state-
> ment that does one thing. By their nature, switch statements always do N things.
Robert C. Martin: Clean Code page 37

Reply:
This is already an improvement over the if-else if... construct in my eyes and it's not the goal to refactor the complete component.
So I'd leave it like that

Reply:
But it's more of a personal opinion and up to you if you want to keep it or not